### PR TITLE
chore: thirdparty rework changelog

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

Updated changelog based on changes done so far and also some of the recent changes discussed.

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,258 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+        AuthorizationEndpointQueryParams: map[string]interface{}{},
+        TokenEndpoint:                    "...",
+        TokenEndpointBodyParams:          map[string]interface{}{},
+        UserInfoEndpoint:                 "...",
+        UserInfoEndpointQueryParams:      map[string]interface{}{},
+        UserInfoMap: tpmodels.TypeUserInfoMap{
+            FromUserInfoAPI: struct {
+                UserId        string "json:\"userId\""
+                Email         string "json:\"email\""
+                EmailVerified string "json:\"emailVerified\""
```
Comment (rishabhpoddar) [inline CHANGELOG.md:168]: Why is it required to give `"json:`..`"` and not just `"userId"`

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,258 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+    - One of redirectURIInfo (for code flow) or oAuthTokens (for token flow) is required
+  - Added: optional `tenantId` to support multi-tenancy
+- Updated `appleRedirectHandlerPOST` 
+  - to accept all the form fields instead of just the code
+  - to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+  - to use HTTP 303 instead of javascript based redirection.
+- Exposing `ManuallyCreateOrUpdateUserWithContext` instead of `SignInUpWithContext` recipe function in the thirdparty, thirdpartyemailpassword and thirdpartypasswordless recipes
+  - `SignInUpWithContext` is used by the SDK only to handle the sign in/up flow. It is not meant to be used by the user.
+  - `ManuallyCreateOrUpdateUserWithContext` recipe function is added to allow the user to manually create or update a user in the core.
+
```
Comment (rishabhpoddar) [inline CHANGELOG.md:34]: Also mention recipe interface changes

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sattvikc) [reply CHANGELOG.md:34]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (sattvikc) [reply CHANGELOG.md]: Will create a new type for this so that it simplifies the user input. Have updated the changelog according to that.

Comment (rishabhpoddar) [reply CHANGELOG.md]: why?? can't it just use the same as the other map?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,253 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+    - One of redirectURIInfo (for code flow) or oAuthTokens (for token flow) is required
+  - Added: optional `tenantId` to support multi-tenancy
+- Updated `appleRedirectHandlerPOST` 
+  - to accept all the form fields instead of just the code
+  - to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+  - to use HTTP 303 instead of javascript based redirection.
+- Exposing `ManuallyCreateOrUpdateUserWithContext` instead of `SignInUpWithContext` recipe function in the thirdparty, thirdpartyemailpassword and thirdpartypasswordless recipes
+  - `SignInUpWithContext` is used by the SDK only to handle the sign in/up flow. It is not meant to be used by the user.
+  - `ManuallyCreateOrUpdateUserWithContext` recipe function is added to allow the user to manually create or update a user in the core.
+  - Recipe changes:
```
Comment (rishabhpoddar) [inline CHANGELOG.md:34]:
```suggestion
  - Recipe interface changes:
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sattvikc) [reply CHANGELOG.md]: We discussed this on call. Created a new type to avoid inline struct.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**